core/frontend/src/internal/tile/map/ImageryProviders/WmtsMapLayerImageryProvider.ts

```diff
@@ -51,16 +59,48 @@ export class WmtsMapLayerImageryProvider extends MapLayerImageryProvider {
       }
     }
   }
+
+  public override get maximumZoomLevel(): number { return this._maximumZoomLevel; }
+
   private initDisplayedLayer() {
-    if (0 === this._settings.subLayers.length) {
-      assert(false);
+    if (0 === this._settings.subLayers.length && this._capabilities?.contents?.layers && this._capabilities.contents.layers.length > 0) {
```

Comment:
Since you're referring to `this._capabilities.contents.layers` a few times, setting it to a var would improve readability.

Reply:
I agree, fixed.